src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -199,6 +211,7 @@ await Task.WhenAny(sendRequestTask, readResponseTask).ConfigureAwait(false) == s
                         // Exceptions will be bubbled up from sendRequestTask here,
                         // which means the result of readResponseTask won't be observed directly:
                         // Do a background await to log any exceptions.
+                        _concurrentRead = true;
```

Comment:
> but might not return the buffers to the pool in all cases (timing dependent)

I assume one of those cases (and the most probable one?) is when `readResponseTask` is completed by this point? Do I get it right that it is still a very rare case? Shall we call it out in comments that this could happen?

Reply:
Or it doesn't see the change of the value yet, or is just past the check, but yeah, in general this is it.

Reply:
Added comment.